bin/java

```diff
@@ -213,7 +213,7 @@ _get_heroku_version() {
 }
 
 _get_url_status() {
-  curl --retry 3 --silent --head -w "%{http_code}" -L "${1}" -o /dev/null
+  curl --retry 3 --retry-connrefused --connect-timeout 5 --silent --head -w "%{http_code}" -L "${1}" -o /dev/null
```

Comment:
`--fail` is intentionally omitted here, since the whole point of this function is to return the HTTP status code (even if it's not 200)

Reply:
Awesome, I was checking for this use-case in all PRs. Great to see that you're already took care to check for this! :)